lib/countly.js

```diff
@@ -1797,6 +1797,236 @@
             }
         }
     };
+
+    /**
+    * Initialize feedbacks
+    * @param {Function} callback - Callback function
+    **/
+    Countly.get_available_feedback_widgets = function(callback) {
+        if (!Countly.check_consent("feedback")) {
+            if (callback) {
+                callback(null, new Error("Consent for feedback not provided."));
+            }
+
+            return;
+        }
+
+        if (offlineMode) {
+            log("Cannot enable feedback in offline mode.");
+            return;
+        }
+
+        var url = Countly.url + readPath;
+        var data = {
+            method: "feedback",
+            device_id: Countly.device_id,
+            app_key: Countly.app_key
+        };
+
+        sendXmlHttpRequest(url, data, function(err, params, responseText){
+            if(err) {
+                log("Error occured while fetching feedbacks", err);
+                if (callback) {
+                    callback(null, err);
+                }
+
+                return;
+            }
+
+            try {
+                var response = JSON.parse(responseText);
+                var feedbacks = response.result || [];
+                if (callback) {
+                    callback(feedbacks, null);
+                }
+
+                return;
+            } catch (error) {
+                log("Error while processing feedbacks", error);
+                if (callback) {
+                    callback(null, error);
+                }
+
+                return;
+            }
+        });
+    };
+
+    /**
+    * Present the feedback widget in webview
+    * @param {Object} presentableFeedback - Current presentable feedback
+    **/
+    Countly.present_feedback_widget = function (presentableFeedback) {
+        if (!Countly.check_consent("feedback")) {
+            if (callback) {
+                callback(null, new Error("Consent for feedback not provided."));
```

Comment:
Codacy found an issue: ['callback' is not defined.](https://app.codacy.com/manual/ar2rsawseen/countly-sdk-web/pullRequest?prid=6418608)